nautobot/extras/views.py

```diff
@@ -949,9 +949,12 @@ def _get_job_model_or_404(self, class_path=None, slug=None):
 
     def get(self, request, class_path=None, slug=None):
         job_model = self._get_job_model_or_404(class_path, slug)
-
+        template_name = "extras/job.html"
         try:
-            job_form = job_model.job_class().as_form(initial=normalize_querydict(request.GET))
+            job_cls = job_model.job_class()
+            job_form = job_cls.as_form(initial=normalize_querydict(request.GET))
+            if hasattr(job_cls, "template_name"):
+                template_name = job_cls.template_name
```

Comment:
Doesn't hurt to be verbose.
```suggestion
            job_class = job_model.job_class()
            job_form = job_class.as_form(initial=normalize_querydict(request.GET))
            if hasattr(job_class, "template_name"):
                template_name = job_class.template_name
```